### PR TITLE
Convert to QSimCircuit after resolving params.

### PR DESCRIPTION
**Pull request by 95-martin-orion**
This also resolves a long-standing question of whether qsim supports parameterized circuits: now it does!

### CODE REVIEW

**Comment by 95-martin-orion** [comment]
TODO(@95-martin-orion): cut a new version and release qsimcirq after this PR. (May wait for a couple other changes as well)

**Review by sergeisakov — APPROVED** [review]
Should `_resolve_parameters_` be removed from qsim_circuit.py? It seems it is not used now and potentially it could lead to bugs.

**Comment by 95-martin-orion** [comment]
> Should `_resolve_parameters_` be removed from qsim_circuit.py? It seems it is not used now and potentially it could lead to bugs.

This is a good point - we really should handle the `_resolve_parameters_` return type in the qsim implementation. Restructured the PR to handle it that way instead - please take a look.

File: qsimcirq/qsim_circuit.py [hunk 1]
```diff
@@ -127,11 +127,8 @@ def __eq__(self, other):
 
   def _resolve_parameters_(self, param_resolver: cirq.study.ParamResolver):
 
-    qsim_circuit = super()._resolve_parameters_(param_resolver)
-
-    qsim_circuit.device = self.device
-
-    return qsim_circuit
+    return QSimCircuit(
```
Comment (sergeisakov) [inline qsimcirq/qsim_circuit.py:130]: Looks good to me. Perhaps empty line 129 should be deleted.

**Review by sergeisakov — APPROVED** [review]

File: qsimcirq/qsim_circuit.py (hunk 1 above)
Comment (95-martin-orion) [reply qsimcirq/qsim_circuit.py:129]: Done.

**Pull request merged**